flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/SqlGatewayServiceStatementITCase.java

```diff
@@ -52,18 +59,49 @@
 /** Test {@link SqlGatewayService}#executeStatement. */
 public class SqlGatewayServiceStatementITCase extends AbstractSqlGatewayStatementITCase {
 
-    private final SessionEnvironment defaultSessionEnvironment =
+    private static final SessionEnvironment DEFAULT_SESSION_ENVIRONMENT =
+            SessionEnvironment.newBuilder()
+                    .setSessionEndpointVersion(MockedEndpointVersion.V1)
+                    .build();
+
+    private static final SessionEnvironment SESSION_ENVIRONMENT_WITH_PLAN_CACHE_ENABLED =
             SessionEnvironment.newBuilder()
                     .setSessionEndpointVersion(MockedEndpointVersion.V1)
+                    .addSessionConfig(
+                            Collections.singletonMap(
+                                    SQL_GATEWAY_SESSION_PLAN_CACHE_ENABLED.key(), "true"))
                     .build();
 
     private SessionHandle sessionHandle;
 
+    @Parameters(name = "parameters={0}")
+    public static List<TestParameters> parameters() throws Exception {
+        return listFlinkSqlTests().stream()
+                .map(path -> new StatementTestParameters(path, path.endsWith("repeated_dql.q")))
+                .collect(Collectors.toList());
+    }
+
     @BeforeEach
     @Override
     public void before(@TempDir Path temporaryFolder) throws Exception {
         super.before(temporaryFolder);
-        sessionHandle = service.openSession(defaultSessionEnvironment);
+        SessionEnvironment sessionEnvironment =
+                isPlanCacheEnabled()
+                        ? SESSION_ENVIRONMENT_WITH_PLAN_CACHE_ENABLED
+                        : DEFAULT_SESSION_ENVIRONMENT;
+        sessionHandle = service.openSession(sessionEnvironment);
+    }
+
+    @AfterEach
+    public void after() {
+        if (isPlanCacheEnabled()) {
+            CacheStats cacheStats =
+                    ((SqlGatewayServiceImpl) service)
+                            .getSession(sessionHandle)
+                            .getPlanCacheManager()
+                            .getCacheStats();
+            assertThat(cacheStats).isEqualTo(new CacheStats(4, 14, 0, 0, 0, 0));
```

Comment:
The design of this test is very fragile, since adding more tests in `*.iq` will all break this test, and they would be forced to change this test expectation as well. I'll think again about how we can test it.

Reply:
This assertion affects `repeated_dql.q` as we only enable plan cache for it. So we only have to modify the expected value when we change the queries in `repeated_dql.q`.